ircstates/server.py

```diff
@@ -905,3 +942,49 @@ def _handle_loggedout(self, line: Line) -> Emit:
         self.account = None
         self._self_hostmask(hostmask)
         return self._emit()
+
+    @line_handler('*')
+    def _handle_all(self, line: Line) -> None:
+        if not any(cap in self.agreed_caps for cap in ('solanum.chat/oper', 'solanum.chat/realhost')):
+            return
+        
+        try:
+            hostmask = line.hostmask
```

Comment:
instead you can check if `line.source` is not None

Reply:
done